src/components/BooleanPill/BooleanPill.tsx

```diff
@@ -0,0 +1,69 @@
+import { Icon } from '@virtuslab/tetrisly-icons';
+import { MouseEventHandler, useCallback, useMemo, type FC } from 'react';
+
+import { BooleanPillProps } from './BooleanPill.props';
+import { stylesBuilder } from './stylesBuilder';
+import { Avatar } from '../Avatar';
+
+import { tet } from '@/tetrisly';
+
+export const BooleanPill: FC<BooleanPillProps> = ({
+  state = 'default',
+  isInverted = false,
+  avatar,
+  text,
+  custom,
+  onChange,
+  ...rest
+}) => {
+  const styles = useMemo(
+    () => stylesBuilder(state, isInverted, custom),
+    [custom, isInverted, state],
+  );
+
+  const avatarProps = useMemo(
+    () =>
+      avatar &&
+      ('image' in avatar
+        ? {
+            img: { src: avatar.image, alt: 'avatar' },
+            appearance: 'image' as const,
+          }
+        : {
+            initials: avatar.initials,
+            appearance: avatar.appearance,
+          }),
+
+    [avatar],
+  );
+
+  const handleOnClick: MouseEventHandler<HTMLSpanElement> = useCallback(() => {
+    if (state !== 'disabled') {
+      onChange?.(state === 'default');
+    }
+  }, [onChange, state]);
+
+  return (
+    <tet.span
+      data-state={state}
+      onClick={handleOnClick}
+      data-testid="boolean-pill"
```

Comment:
Tou should add here some default `tabIndex` value to handle `:focus` and `:active` states.